atom/browser/lib/power_observer_linux.cc

```diff
@@ -16,6 +16,10 @@ const char kLogindServiceName[] = "org.freedesktop.login1";
 const char kLogindObjectPath[] = "/org/freedesktop/login1";
 const char kLogindManagerInterface[] = "org.freedesktop.login1.Manager";
 
+// Store shutdown lock as a global, since we only want to release it when the
+// main process exits.
+base::ScopedFD shutdown_lock;
```

Comment:
I could move `shutdown_lock` into the class, but it is possible that the `ScopedFD` goes out of scope before the application has fully exited, which results in a small probability of systemd killing the app.

Maybe it is not a problem in practice since this class instance is probably deleted only a bit sooner than the app exits, but we would loose guarantee that the app exits cleanly by itself.